### PR TITLE
Some janitoring

### PR DESCRIPTION
**Pull request by akva2**
Quell some warnings still being emitted when bulding with clang.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this serial please

**Comment by akva2** [comment]
jenkins build this serial please

File: opm/simulators/linalg/ISTLSolverEbos.hpp [hunk 1]
```diff
@@ -650,8 +650,8 @@ class WellModelMatrixAdapter : public Dune::AssembledLinearOperator<M,X,Y>
         void makeOverlapRowsInvalid(Matrix& ebosJacIgnoreOverlap) const
         {
             //value to set on diagonal
-            typedef Dune::FieldMatrix<Scalar, numEq, numEq >        MatrixBlockType;
-            MatrixBlockType diag_block(0.0);
```
Comment (GitPaean) [inline opm/simulators/linalg/ISTLSolverEbos.hpp]: why this one?

Comment (GitPaean) [reply opm/simulators/linalg/ISTLSolverEbos.hpp]: I guess it shadows some types somewhere.

Comment (GitPaean) [reply opm/simulators/linalg/ISTLSolverEbos.hpp]: It is good if it still has the `Type` in it to say it is a type name.

Comment (akva2) [reply opm/simulators/linalg/ISTLSolverEbos.hpp]: Yes it shadows. Since the typedef is not reused, i removed it.

**Comment by akva2** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
Thanks for the efforts.  Jenkins is happy and it is going in now.

**Pull request merged**